Tests/test_energy.gd

```diff
@@ -0,0 +1,60 @@
+extends TestBase
+
+var card: CardBase
+
+# @overide
+func before_each():
+	super()
+	_player_energy_component.ignore_cost = false
+	_player_energy_component.energy = 0
+	_player_energy_component.energy_generation = 3
+	_player_energy_component.Max_energy = 4
+
+func test_add_energy():
+	card = load("res://Cards/Resource/Card_Energy.tres") # adds 70 energy
+
+	card.on_card_play(_player, [_player])
+	assert_eq(_player_energy_component.energy, 70)
+
+
+func test_using_energy():
+	_player_energy_component.add_energy(10)
+	card = load("res://Cards/Resource/Card_Damage.tres") # costs 7 by default
+	card.energy.energy_cost = 7
+
+	card.on_card_play(PlayerManager.player, [_enemy])
+
+	assert_eq(_player_energy_component.energy, 3)
+
+func test_no_energy():
+
+	_card_container.draw_cards(1)
+
+	var _queued_card: CardWorld = _card_container.cards_in_hand[0]
+	_queued_card.card_data.energy.energy_cost = 999
+
+	_card_container._on_card_clicked(_queued_card)
+
+	assert_false(_card_container.is_card_queued())
+
+func test_having_energy():
+
+	_player_energy_component.add_energy(10)
```

Comment:
I notice that you always use energy amounts that the player won't usually go to
it would be good to have some functions that test for energy levels that would happen more usually